src/blueprint/telemetry.rs

```diff
@@ -57,6 +58,8 @@ pub fn to_opentelemetry<'a>() -> TryFold<'a, ConfigModule, Telemetry, String> {
                 config::TelemetryExporter::Prometheus(config) => {
                     Valid::succeed(TelemetryExporter::Prometheus(config.clone()))
                 }
+                config::TelemetryExporter::Apollo(apollo) => validate_apollo(apollo.clone())
+                    .and_then(|apollo| Valid::succeed(TelemetryExporter::Apollo(apollo))),
```

Comment:
Add a test for execution spec where we are validating the error with a trace pointing to where the error is.